aries-site/src/examples/templates/dashboards/components/Legend.js

```diff
@@ -1,30 +1,57 @@
-import { useContext } from 'react';
+import { Fragment, useContext, useState } from 'react';
 import PropTypes from 'prop-types';
-import { Box, List, Text, ThemeContext } from 'grommet';
+import { Box, Button, List, Text, ThemeContext } from 'grommet';
 
-export const Legend = ({ values, ...rest }) => {
+export const Legend = ({ mode, onClick, values, ...rest }) => {
+  const [active, setActive] = useState();
   const theme = useContext(ThemeContext);
   const size = theme.global.edgeSize.small;
 
+  const Wrapper = onClick ? Button : Fragment;
+
   return (
     <List
       data={values}
       defaultItemProps={{ pad: { vertical: 'xxsmall' } }}
       {...rest}
     >
       {datum => (
-        <Box direction="row" gap="small" justify="between">
-          <Box direction="row" align="center" gap="xsmall">
-            <Box background={datum.color} height={size} width={size} round />
-            <Text>{datum.label}</Text>
+        <Wrapper
+          {...(onClick
+            ? {
+                onClick: () => {
+                  onClick(datum);
+                  setActive(active !== datum ? datum : undefined);
+                },
+                active: active === datum,
+                hoverIndicator: true,
+              }
+            : {})}
+        >
+          <Box
+            direction="row"
+            gap="small"
+            justify="between"
+            pad={
+              onClick
```

Comment:
Should we consider adding a similar onClick conditional to the `defaultItemProps={{ pad: { vertical: 'xxsmall' } }}`? Set it to 'none' when onClick? This would keep Legend spacing consistent.

Reply:
Yes, I'll do that.